src/Wallets/WalletAccount.cs

```diff
@@ -0,0 +1,126 @@
+using System.Collections.Immutable;
+using Concordium.Sdk.Crypto;
+using Concordium.Sdk.Transactions;
+using Concordium.Sdk.Types;
+using Newtonsoft.Json;
+
+namespace Concordium.Sdk.Wallets;
+
+/// <summary>
+/// Represents an account imported from one of the supported wallet export formats.
+///
+/// In particular it supports import of the browser and genesis wallet key export
+/// formats. The class implements <see cref="ITransactionSigner"/> so it may be used
+/// for signing transactions.
+///
+/// This structure does not have the encryption key for sending encrypted transfers, it only
+/// contains keys for signing transactions.
+/// </summary>
+public class WalletAccount : ITransactionSigner
+{
+    /// <summary>
+    /// The address of the imported account.
+    /// </summary>
+    public AccountAddress AccountAddress { get; init; }
+
+    /// <summary>
+    /// Internal representation of a signer.
+    /// </summary>
+    private readonly TransactionSigner _signer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="WalletAccount"/> class.
+    /// </summary>
+    /// <param name="accountAddress">The address of the imported account.</param>
+    /// <param name="signKeys">The signers corresponding to the keys of the imported account.</param>
+    private WalletAccount(
+        AccountAddress accountAddress,
+        Dictionary<AccountCredentialIndex, Dictionary<AccountKeyIndex, ISigner>> signKeys
+    )
+    {
+        this.AccountAddress = accountAddress;
+        this._signer = new TransactionSigner();
+        signKeys
+            .ToList()
```

Comment:
Could you avoid the ToList()?

Reply:
I'm not sure. I wanted some representation that I could map over, but there can at most be 255 entries anyways.